Source/Core/Core/AchievementManager.h

```diff
@@ -61,6 +66,19 @@ class AchievementManager
   rc_api_fetch_game_data_response_t m_game_data{};
   bool m_is_game_loaded = false;
 
+  struct UnlockStatus
+  {
+    AchievementId game_data_index;
+    enum class UnlockType
+    {
+      LOCKED,
+      SOFTCORE,
+      HARDCORE
+    } remote_unlock_status = UnlockType::LOCKED;
+    int session_unlock_count = 0;
+  };
+  std::map<AchievementId, UnlockStatus> m_unlock_map;
```

Comment:
Unless you need the `std::map` property that it's ordered by keys when iterating, please use `std::unordered_map` instead.

Reply:
Today I learned that unordered_map lookups faster than map.